config/samples/dataplane_v1beta1_openstackdataplane_nmstate.yaml

```diff
@@ -0,0 +1,266 @@
+      nodeTemplate:
+        # Defining the novaTemplate here means the nodes in this role are
+        # computes
+        nova: {}
+        managementNetwork: ctlplane
+        ansibleUser: root
+        ansiblePort: 22
+        ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+        ansibleVars: |
+          service_net_map:
+            nova_api_network: internal_api
+            nova_libvirt_network: internal_api
+
+          # These vars are for the network config templates themselves and are
+          # considered EDPM network defaults.
+          neutron_physical_bridge_name: br-ex
+          neutron_public_interface_name: eth0
+
+          # edpm_nodes_validation
+          edpm_nodes_validation_validate_controllers_icmp: false
+          edpm_nodes_validation_validate_gateway_icmp: false
+
+          edpm_ovn_metadata_agent_default_transport_url: rabbit://default_user@rabbitmq.openstack.svc:5672
+          edpm_ovn_metadata_agent_metadata_agent_ovn_ovn_sb_connection: tcp:10.217.5.121:6642
+          edpm_ovn_metadata_agent_metadata_agent_default_nova_metadata_host: 127.0.0.1
+          edpm_ovn_metadata_agent_metadata_agent_default_metadata_proxy_shared_secret: 12345678
+          edpm_ovn_metadata_agent_default_bind_host: 127.0.0.1
+          edpm_chrony_ntp_servers:
+          - clock.redhat.com
+          - clock2.redhat.com
+
+          ctlplane_dns_nameservers:
+          - 192.168.122.1
+          dns_search_domains: []
+          edpm_ovn_dbs:
+          - 192.168.24.1
+
+          edpm_ovn_controller_agent_image: quay.io/podified-antelope-centos9/openstack-ovn-controller:current-podified
+          edpm_iscsid_image: quay.io/podified-antelope-centos9/openstack-iscsid:current-podified
+          edpm_logrotate_crond_image: quay.io/podified-antelope-centos9/openstack-cron:current-podified
+          edpm_nova_compute_container_image: quay.io/podified-antelope-centos9/openstack-nova-compute:current-podified
+          edpm_nova_libvirt_container_image: quay.io/podified-antelope-centos9/openstack-nova-libvirt:current-podified
+          edpm_ovn_metadata_agent_image: quay.io/podified-antelope-centos9/openstack-neutron-metadata-agent-ovn:current-podified
+
+          gather_facts: false
+          enable_debug: false
+          # edpm firewall, change the allowed CIDR if needed
+          edpm_sshd_configure_firewall: true
+          edpm_sshd_allowed_ranges: ['192.168.122.0/24']
+          # SELinux module
+          edpm_selinux_mode: enforcing
+          edpm_hosts_entries_undercloud_hosts_entries: []
+          # edpm_hosts_entries role
+          edpm_hosts_entries_extra_hosts_entries:
+          - 172.17.0.80 glance-internal.openstack.svc neutron-internal.openstack.svc cinder-internal.openstack.svc nova-internal.openstack.svc placement-internal.openstack.svc keystone-internal.openstack.svc
+          - 172.17.0.85 rabbitmq.openstack.svc
+          - 172.17.0.86 rabbitmq-cell1.openstack.svc
```

Comment:
when using install_yamls, dns is now deployed and available per default on 192.168.122.80 [1]. the internal api endpoints can be resolved using dns instead of adding those hosts entries. Should we change this PR to use 192.168.122.80 as the `ctlplane_dns_nameservers` and remove adding these hosts entries?

[1] https://github.com/openstack-k8s-operators/infra-operator/blob/main/config/samples/network_v1beta1_dnsmasq.yaml

Reply:
It sounds good! I haven't thought about that, let me fix the CR.

Reply:
@stuggi I'll add the dns as `network_state.dns_resolver.config.server` for each compute node host vars since `ctlplane_dns_nameservers` is used by os-net-config.